src/wallet/wallet.cpp

```diff
@@ -2723,12 +2723,18 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
                 continue;
             }
         }
+        // Unordered transaction inputs
+        txNew.vin.clear();
+        std::vector<pair<const CWalletTx*,unsigned int>> selectedCoins(setCoins.begin(), setCoins.end());
+        random_shuffle(selectedCoins.begin(), selectedCoins.end(), GetRandInt);
```

Comment:
random_shuffle() is obsoleted in c++17.
If that is not a problem, this looks good to me.

Reply:
Okay, changing it for `std::shuffle`. We need a pseudo-random number engines and I am not sure if `std::default_random_engine` is enough ([C++ Reference](https://www.cplusplus.com/reference/random/)).

Reply:
Could possibly backport Shuffle from bitcoin PR #14624

Reply:
Might as well use `std::shuffle` with the `std::default_random_engine` with time-based seed in this PR and open a new issue for refactoring the shuffle function and the inputs format?

Reply:
I'm generally in favour of "If it improves the state of things, lets get it in, worry about perfect later" (but noting I have high standards for improving :) )

Reply:
To clarify; I mean I like how this is, although targetting post 1.14.4